src/pal/src/loader/module.cpp

```diff
@@ -676,7 +676,7 @@ PAL_RegisterModule(
 {
     HINSTANCE hinstance = nullptr;
 
-    int err = PAL_InitializeDLL();
+    int err = PAL_InitializeDLLWithoutSEH();
```

Comment:
It should not break the debugger libs since they are not loaded via LoadLibrary but directly via dlopen and they call the PAL_InitializeDll themselves.
However, I agree that this PAL_RegisterModule thing is not needed for anything and just complicates stuff. It that's removed, then calling the PAL_InitializeDllWithoutSEH from the DllMain of the shared library should work well. We would need to add DllMain to the jitter shared library.